br/pkg/restore/client.go

```diff
@@ -2543,15 +2614,20 @@ func (rc *Client) RestoreMetaKVFiles(
 	filesInDefaultCF = SortMetaKVFiles(filesInDefaultCF)
 	filesInWriteCF = SortMetaKVFiles(filesInWriteCF)
 
-	// Preconstruct the map and save it into external storage.
-	if err := rc.PreConstructAndSaveIDMap(
-		ctx,
-		filesInWriteCF,
-		filesInDefaultCF,
-		schemasReplace,
-	); err != nil {
-		return errors.Trace(err)
+	if schemasReplace.NeedConstructIdMap() {
+		// Preconstruct the map and save it into external storage.
+		if err := rc.PreConstructAndSaveIDMap(
+			ctx,
+			filesInWriteCF,
+			filesInDefaultCF,
+			schemasReplace,
+		); err != nil {
+			return errors.Trace(err)
+		}
 	}
+	failpoint.Inject("failed-after-id-maps-saved", func(_ failpoint.Value) {
+		failpoint.Return(errors.New("failpoint: failed after id maps saved"))
+	})
```

Comment:
It would be more precise if the failpoint is inside RestoreMetaKVFilesWithBatchMethod or UpdateSchemaVersion

Reply:
done